crawl-ref/source/travel.cc

```diff
@@ -210,8 +210,12 @@ static inline bool _is_safe_cloud(const coord_def& c)
 // This is done, so traps etc. will usually be circumvented where possible.
 static inline int _feature_traverse_cost(dungeon_feature_type feature)
 {
-    if (feature == DNGN_SHALLOW_WATER || feat_is_closed_door(feature))
+    if ((feature == DNGN_SHALLOW_WATER && !player_likes_water(true)
```

Comment:
This parenthesization looks wrong to me. You've written "If it's shallow water and you don't like water, or if you're a merfolk regardless of what kind of feature it is". From your commit description it sounds like you want to parenthesize the && clause to include the merfolk check.